solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java

```diff
@@ -157,6 +175,36 @@ public ShardHandler getShardHandler(final HttpClient httpClient){
     return new HttpShardHandler(this, httpClient);
   }
 
+  /**
+   * Returns this Factory's {@link WhitelistHostChecker}.
+   * This method can be overridden to change the checker implementation.
+   */
+  public WhitelistHostChecker getWhitelistHostChecker() {
+    return this.whitelistHostChecker;
+  }
+
+  @Deprecated // For temporary use by the TermsComponent only.
+  static boolean doGetDisableShardsWhitelist() {
+    return getDisableShardsWhitelist();
+  }
+
```

Comment:
Custom "getShardsWhitelist" method. Previously the value was being retrieved only from configuration settings as a property set for a bean, which got difficult fast if more than one configuration file has been specified.

I think this fix should allow for keeping the value empty, setting the default value across all HttpShardHandlerFactory implementations created and overriding the default value in a specific config if needed.